### PR TITLE
Allow reading TimeSpan from INTEGER/REAL

### PR DESCRIPTION
**Pull request by AlexanderTaeschner**
Adds GetTimeSpan to SqliteDataReader and implements conversion from REAL to TimeSpan (assumed is that the value represents the fractional number of days (e.g. as result of a subtraction of two Julian date values)).

Addresses #411

### CODE REVIEW

File: src/Microsoft.Data.Sqlite.Core/SqliteValueReader.cs [hunk 1]
```diff
@@ -90,10 +90,31 @@ public virtual Guid GetGuid(int ordinal)
...
+                    var days = (int)span;
+                    span = (span - days) * 24;
+                    var hours = (int)span;
+                    span = (span - hours) * 60;
+                    var minutes = (int)span;
+                    span = (span - minutes) * 60;
+                    var seconds = (int)span;
+                    span = (span - seconds) * 1000;
+                    var milliseconds = (int)span;
+                    return new TimeSpan(days, hours, minutes, seconds, milliseconds);
```
Comment (bricelam) [inline src/Microsoft.Data.Sqlite.Core/SqliteValueReader.cs]: Just use `TimeSpan.FromDays()`?

File: src/Microsoft.Data.Sqlite.Core/SqliteValueReader.cs [hunk 2]
```diff
@@ -90,10 +90,31 @@ public virtual Guid GetGuid(int ordinal)
...
-            => IsDBNull(ordinal)
-                ? GetNull<TimeSpan>()
-                : TimeSpan.Parse(GetString(ordinal));
+        public virtual TimeSpan GetTimeSpan(int ordinal)
+        {
+            var sqliteType = GetSqliteType(ordinal);
+            switch (sqliteType)
+            {
+                case raw.SQLITE_NULL:
+                    return GetNull<TimeSpan>();
```
[the remark this answers is not in the document]
Comment (AlexanderTaeschner) [reply src/Microsoft.Data.Sqlite.Core/SqliteValueReader.cs]: CreateFunction_works_when_parameter_null_and_type_TimeSpan fails without this.

File: src/Microsoft.Data.Sqlite.Core/SqliteValueReader.cs (hunk 1 above)
Comment (AlexanderTaeschner) [reply src/Microsoft.Data.Sqlite.Core/SqliteValueReader.cs]: Done, this is really much simpler.

File: src/Microsoft.Data.Sqlite.Core/SqliteValueReader.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (bricelam) [reply src/Microsoft.Data.Sqlite.Core/SqliteValueReader.cs]: That test is changing in that PR too 😄

**Comment by AlexanderTaeschner** [comment]
Removef special casing for SQLITE_Null now. PR is blocked until #414 landed.

**Comment by AlexanderTaeschner** [comment]
Rebased on top of current head (including #414).

**Comment by bricelam** [comment]
Thanks. For rebasing. 😄 lol, I was in the middle of doing it for you...

**Pull request merged**